sql/catalyst/src/test/scala/org/apache/spark/sql/types/TestUDT.scala

```diff
@@ -58,4 +63,22 @@ object TestUDT {
 
     override def equals(other: Any): Boolean = other.isInstanceOf[MyDenseVectorUDT]
   }
+
+  private[sql] class MyXMLGregorianCalendarUDT extends UserDefinedType[XMLGregorianCalendar] {
```

Comment:
Can you use the existing UDT for testing this pr instead of defining the new UDT?

Reply:
No, this isn't possible since we need to override the `jvalue` to the native DataType type. Otherwise, it will be caught by: https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructType.scala#L592